pubtools/_quay/operator_pusher.py

```diff
@@ -547,39 +547,75 @@ def build_index_images(self):
             is_advisory_source = all(
                 [re.match(r"^[A-Z0-9:\-]{4,40}$", item.origin) for item in non_fbc_items]
             )
-            item_groups = {}
+            item_groups = {
+                version: {
+                    "items": [],
+                    "overwrite": True,
+                    "build_tags": [],
+                    "destination_tags": [version],
+                }
+            }
             if is_hotfix and not is_advisory_source:
                 raise ValueError("Cannot push hotfixes without an advisory")
+
             if is_hotfix:
                 for item in non_fbc_items:
-                    item_groups.setdefault(item.origin, []).append(item)
+                    hotfix_tag = "{0}-{1}-{2}".format(
+                        version,
+                        item.metadata["com.redhat.hotfix"],
+                        item.origin.split("-")[1].replace(":", "-"),
+                    )
+                    item_groups.setdefault(
+                        item.origin,
+                        {
+                            "items": [],
+                            "overwrite": False,
+                            "build_tags": [],
+                            "destination_tags": [hotfix_tag],
+                        },
+                    )
+                    item_groups[item.origin]["items"].append(item)
             else:
-                item_groups["default"] = non_fbc_items
+                for item in non_fbc_items:
+                    if item.metadata.get("com.redhat.pre-release"):
+                        item_groups.setdefault(
+                            f'{version}.{item.metadata.get("com.redhat.pre-release")}',
+                            {
+                                "items": [],
+                                "build_tags": [
+                                    f"{version}.{item.metadata.get('com.redhat.pre-release')}"
+                                ],
+                                "overwrite": False,
+                                "destination_tags": [
```

Comment:
destination tag for pre-release bundle should also use "version-label-adv-id" format

Reply:
ah, I see you already picked this in the document. But that would limit source for pre-releases only to advisory. Which is not necessary

Reply:
I see, AFAIK pre-release content shipping is planned to be on the same lines as hotfix releases, which would involve releasing the build via the ET. 
May I ask if you can share more information or references that concludes pre-release content can be shipped via other ways?

Reply:
If 2 product teams end up using the same value for `com.redhat.pre-release` label, that would mean the old index with version-labelvalue tag would be replaced by the new index

Reply:
hmm, good point